### DISCUSSION TITLE
Oh No!  Corrupt file

### DISCUSSION THREAD

**Discussion by BigHonkinDaddy** [powertab/powertabeditor#459] [General]
I was working on my latest project in PowerTab (V2.0.0 alpha19) when my computer froze.  i had to force a reboot and now I can't open my powertab file (.pt2).   I get "Error Opening File: gzip error: iostream stream error".  My last back up of the file is about 3 weeks ago, so kinda bummed I've gotta start all over.  Any chance it can be salvaged?

**Comment by cameronwhite** [comment]
If you're able to attach it here (or send it directly to me) I can poke around and see what data is in there! It's basically a JSON file, but compressed with GZIP to reduce file sizes, so the success probably depends on whether gzip can extract much.

**Reply by BigHonkinDaddy** [reply]
Thanks.  I sent it to your email since I wasn't able to upload the pt2 file type here... and I didn't want to zip it for fear of further complicating the problem with it.

**Reply by cameronwhite** [reply]
Thanks! Unfortunately, it looks like all of the bytes in the file are just zero so there isn't anything valid / recoverable :( 

For the next release #392 is scheduled to be worked on, so hopefully some autosave / auto-backup option might help avoid this. I'm not entirely sure what caused your corrupt file in the first place though! If the freeze happened while saving and caused something incomplete to be saved that might make sense, otherwise I'm not really sure

**Reply by BigHonkinDaddy** [reply]
Well thanks any way for having a look.  The laptop I was using is almost literally a piece of junk. 

Auto-save will be a nice feature, but wouldn't have helped in my case.  I had been saving frequently. The crash/forced-reboot overwrote the file with something else, I guess.  

Thanks for trying and thanks for all the great work you put in to PowerTab.

**Reply by cameronwhite** [reply]
No problem!
One of my thoughts with auto-save was to have some option to save copies of the file somewhere, so you e.g. might have access to the last few versions of the file. That way if you do happen to mess up the main file you have some history to fall back to
